contracts/contracts/validator-registry/rewards/RewardManager.sol

```diff
@@ -0,0 +1,202 @@
+        unclaimedRewards[msg.sender] = 0;
+        (bool success, ) = payable(msg.sender).call{value: amount}("");
+        if (!success) {
+            revert RewardsClaimFailed();
+        }
+        emit RewardsClaimed(msg.sender, amount);
+    }
+
+    /// @dev Allows the owner to claim orphaned rewards to appropriate addresses.
+    function claimOrphanedRewards(bytes[] calldata pubkeys, address toPay) external onlyOwner {
+        uint256 totalAmount = 0;
+        uint256 len = pubkeys.length;
+        for (uint256 i = 0; i < len; ++i) {
+            bytes calldata pubkey = pubkeys[i];
+            uint256 amount = orphanedRewards[pubkey];
+            require(amount > 0, NoOrphanedRewards());
+            orphanedRewards[pubkey] = 0;
+            totalAmount += amount;
+        }
+        (bool success, ) = payable(toPay).call{value: totalAmount}("");
+        if (!success) {
+            revert OrphanedRewardsClaimFailed();
+        }
+    }
+
+    /// @dev Allows the owner to set the vanilla registry address.
+    function setVanillaRegistry(address vanillaRegistry) external onlyOwner {
+        _setVanillaRegistry(vanillaRegistry);
+    }
+
+    /// @dev Allows the owner to set the mev commit avs address.
+    function setMevCommitAVS(address mevCommitAVS) external onlyOwner {
+        _setMevCommitAVS(mevCommitAVS);
+    }
+
+    /// @dev Allows the owner to set the mev commit middleware address.
+    function setMevCommitMiddleware(address mevCommitMiddleware) external onlyOwner {
+        _setMevCommitMiddleware(mevCommitMiddleware);
+    }
+
+    /// @dev Allows the owner to set the auto claim gas limit.
+    function setAutoClaimGasLimit(uint256 autoClaimGasLimit) external onlyOwner {
+        _setAutoClaimGasLimit(autoClaimGasLimit);
+    }
+
+    // solhint-disable-next-line no-empty-blocks
+    function _authorizeUpgrade(address) internal override onlyOwner {}
+
+    function _setVanillaRegistry(address vanillaRegistry) internal {
+        require(vanillaRegistry != address(0), InvalidAddress());
+        _vanillaRegistry = VanillaRegistryStorage(vanillaRegistry);
+        emit VanillaRegistrySet(address(vanillaRegistry), vanillaRegistry);
+    }
+
+    function _setMevCommitAVS(address mevCommitAVS) internal {
+        require(mevCommitAVS != address(0), InvalidAddress());
+        _mevCommitAVS = MevCommitAVSStorage(mevCommitAVS);
+        emit MevCommitAVSSet(address(mevCommitAVS), mevCommitAVS);
+    }
+
+    function _setMevCommitMiddleware(address mevCommitMiddleware) internal {
+        require(mevCommitMiddleware != address(0), InvalidAddress());
+        _mevCommitMiddleware = MevCommitMiddlewareStorage(mevCommitMiddleware);
+        emit MevCommitMiddlewareSet(address(mevCommitMiddleware), mevCommitMiddleware);
+    }
+
+    function _setAutoClaimGasLimit(uint256 autoClaimGasLimit) internal {
+        require(autoClaimGasLimit > 0, InvalidAutoClaimGasLimit());
+        autoClaimGasLimit = autoClaimGasLimit;
```

Comment:
Is `autoClaimGasLimit` in storage actually get's updated? Looks like it assigns the function parameter to itself.

Reply:
Valid, these setter issues should all be fixed now